planner/plansdk/plansdk.go

```diff
@@ -144,7 +144,9 @@ func findBuildablePlan(plans ...*Plan) *Plan {
 	for _, p := range plans {
 		// For now, pick the first buildable plan.
 		if p.Buildable() {
-			return p
+			// Make a copy so that the original plan is not modified.
+			newP := *p
+			return &newP
```

Comment:
I don't know if this matters for this change, but `ShellInitHook` looks like it's the only field that will be copied. You'd need to do a deep copy if you want the pointers/slices/maps to be copied as well.